docs/api-ref.md

```diff
@@ -2854,9 +2854,34 @@ Name | Description
 `email` |  The email address of the user.
 `fullname` | The full name of the user.
 `name` |   The name of the user. This attribute is required when you are creating a `UserItem` instance.
-`site_role` |  The role the user has on the site. This attribute is required with you are creating a `UserItem` instance. The specific roles vary depending upon the version of the REST API.  For example, for version 2.8 and earlier, the `site_role` can be one of the following: `Interactor`, `Publisher`, `ServerAdministrator`, `SiteAdministrator`, `Unlicensed`, `UnlicensedWithPublish`,  `Viewer`, `ViewerWithPublish`, `Guest`.  For REST API 3.0 and later, the `site_role` can be one of the following `Creator`, `Explorer`, `ExplorerCanPublish`, `ReadOnly` *(viewers from previous API versions who do not have v3.0+ viewer permissions)*, `Viewer`, `SiteAdministratorCreator`, `SiteAdministratorExplorer`, `UnlicensedWithPublish`.
+`site_role` |  The role the user has on the site. This attribute is required with you are creating a `UserItem` instance. See *User Roles* below for details.
 `groups` | The groups that the user belongs to. You must run the populate_groups method to add the groups to the `UserItem`.
 
+**User Roles**
+
+The possible user roles (for the `site_role` attribute) vary depending upon the version of the REST API.
+
+For REST API 3.0 and later:
```

Comment:
I'm not sure where this list came from in the first place, but some of them look invalid.

Looking through rest api docs, here are the lists I found:
[3.0+](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_usersgroups.htm#add_user_to_site) - Creator, Explorer, ExplorerCanPublish, SiteAdministratorExplorer, SiteAdministratorCreator, Unlicensed, or Viewer

[<2.8](https://help.tableau.com/v2018.2/api/rest_api/en-us/REST/rest_api_ref.htm#Add_User_to_Site) - Creator, Explorer, ExplorerCanPublish, ReadOnly, SiteAdministratorExplorer, SiteAdministratorCreator, Unlicensed, or Viewer

3.0 is no longer supported by Tableau, but I think we can still leave the second list.